### ISSUE TITLE
Can't load 3.0.1.23

### ISSUE THREAD

**Issue by radkitten79** [hengband/hengband#4748]
I've never had this problem before, all version up until 3.0.1.22 load to main screen when I double click the game executable and play fine.. but for some reason 3.0.1.23 gives an error when the executable is double clicked. I don't see anything in the change log that would cause this, but the error said "Unable to execute". I've re-downloaded 3.0.1.23 twice now, still get same error. Any ideas? I'll just play on 3.0.1.22 until this gets resolved or I'll try a future version.. so strange.

**Comment by Hourier** [comment]
Please tell us the OS and hardware info, especially CPU vendor and model.
The largest change is between 3.0.1.21 and 22, not 22 and 23.

Notes:
Specific "largest change" is that we have discontinued support for CPUs that don't support [AVX](https://en.wikipedia.org/wiki/Streaming_SIMD_Extensions).
If the pre-installed OS is earlier than Windows 7 SP1, or even if SP1, Hengband may become unable to boot.
Furthermore, we confirmed that Windows Vista can't boot Hengband.

**Comment by radkitten79** [comment]
AMD Quad-Core A6-3650
8GB DDR3 system memory
AMD Radeon HD 6530D integrated graphics
Windows 7 64bit fully updated

I'm looking into getting a new laptop soon, so it's probably a conflict with my hardware.. just blew my mind because I've never had this problem running Hengband, Gensouband, Angband, Frogcomposband, Sil-Q, etc..  I play a lot of ascii and tiled rogue-likes without any problems.. I even play tile-based rogue-likes such as ADOM, Caves of Qud, Cataclysm DDA 0.H and Tales of Maj'Eyal.. so I was perplexed when Hengband  3.0.1.23 wouldn't execute. I re-tried 3.0.1.22, 3.0.1.21 and 3.0.1.20 and all of those load and play fine which is also perplexing.. heh.

**Comment by Hourier** [comment]
Thank you for providing the info!
I checked the support status of A6-3650 and found that this CPU doesn't support AVX.
In other traditional rogue-like games, such as Angband, developers has intention to support OLDEST OSs.
(Oldies examples: 80's mainframes, CPUs not supporting instruction set of x87; IEEE754-based floating-point numbers)

However, current Hengband uses newest technologies in source codes, so we want to maximize program execution performance on the newest CPUs.
(AVX2 and AVX512 is too new, and we have other reasons, so we disable them.)
Furthermore, as you may know, Windows 7 is EOL already (Jan 2020), so we hope this gives you one more reason to purchase a new PC.

BTW, I remembered that there were some mistakes in the English release of 3.0.1.22, which may have caused AVX to be disabled.
On the other hand, 3.0.1.22 has crash bug, so please use 3.0.1.21 for a while.

**Comment by radkitten79** [comment]
Yeah I read that Windows 12 should be releasing 1st quarter of 2025, so when I buy a new laptop I'll make sure it is AVX compatible for sure. Hengband has given me many fond memories for several years and it is a variant that I revisit frequently.

You can mark this issue as closed, I have written down the AVX requirement so I know to look for that when making a new laptop purchase. Lastly, I wanted to take this time to express my thanks to you and the other Hengband developers; this rogue-like has given me many years of enjoyment and the fact that you all keep it updated as a labor of love makes fans like me smile.

**Comment by Hourier** [comment]
Thank you for playing Hengband for a long time!
I close this ticket and we hope you will enjoy Hengband more and more.

BTW, if you'd like, why not join the Hengband user community?
Some of the developers, including myself, can read and write English, and we also have a small number of native English-speaking users.
Thank you!
https://discord.gg/8xW6q5SqXY

**Comment by Hourier** [comment]
I re-opened this ticket.
I implemented the logic to show error pop-up message when this software finds CPUs without AVX.
Would you please check the behavior?
(If OS-level initialization fails, pop-up will not be shown.)
[Hengband.zip](https://github.com/user-attachments/files/18384026/Hengband.zip)

1. Download and decompression the attached file
2. Rename existing Hengband.exe to Hengband.exe.bak
3. Replace the file
4. Run

**Comment by Hourier** [comment]
Another user checked the behavior of this issue.
As a result, I regret to say that the check is not correctly run.
It means that OS-level initialization fails when starting Hengband.
I re-close this case. Thank you!
